xapi/xapi.go

```diff
@@ -146,12 +147,14 @@ func composeRequest(req *tipb.SelectRequest, concurrency int) (*kv.Request, erro
 		kvReq.KeyRanges = tablecodec.EncodeIndexRanges(tid, idxID, req.Ranges)
 		if req.OrderBy != nil {
 			kvReq.Desc = *req.OrderBy[0].Desc
-			kvReq.KeepOrder = true
 		}
 	} else {
 		kvReq.Tp = kv.ReqTypeSelect
 		tid := req.GetTableInfo().GetTableId()
 		kvReq.KeyRanges = tablecodec.EncodeTableRanges(tid, req.Ranges)
+		if req.OrderBy != nil {
+			kvReq.Desc = *req.OrderBy[0].Desc
+		}
```

Comment:
Line 155-157 and Line 148-150 are the same. So line 155-157 can move behind the line 158?